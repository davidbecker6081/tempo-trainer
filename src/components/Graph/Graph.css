.graph-container {
  height: 50vh;
  margin-bottom: 4rem;
  width: 60vw;
  background-color: #FFFFFF;
}
