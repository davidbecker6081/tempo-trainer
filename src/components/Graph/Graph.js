import React, { Component } from 'react';
import { VictoryChart, VictoryArea, VictoryTheme, VictoryAxis, VictoryLabel, VictoryBar } from 'victory';
import './Graph.css';

const Graph = ({ dataHelper, range, channelSet }) => {
  const data = dataHelper.filterDataForGraph(channelSet);
  // const tickValues = data.filter(point => range[range.length - 1] / 10;
  return (
    <div>
      <VictoryChart theme={VictoryTheme.material}>
        <VictoryArea
          style={{ data: { fill: '#0000AA' } }}
          data={data}
          barRatio={0.8}
          x="time"
          y={channelSet}
        />
      </VictoryChart>
    </div>
  );
};

export default Graph;
