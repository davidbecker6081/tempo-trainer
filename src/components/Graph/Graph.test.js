import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Graph from './Graph';
import DataHelper from '../DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('Graph', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      channelSet: '',
      dataHelper: {},
      range: [],
    };
  });

  describe('channelSet is defined', () => {
    beforeEach(() => { mockProps.channelSet = 'power'; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.range = [0, 10]; });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
