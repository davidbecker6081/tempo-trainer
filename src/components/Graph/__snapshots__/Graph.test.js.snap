// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph channelSet is defined should render correclty 1`] = `undefined`;

exports[`Graph dataHelper is defined should render correclty 1`] = `undefined`;

exports[`Graph dataHelper is defined should render correclty 2`] = `undefined`;
